### PR TITLE
fix: table truncation

### PR DESCRIPTION
**Pull request by caarlos0**
it was truncating "too late", causing style leaks

refs https://github.com/charmbracelet/glamour/pull/317

### CODE REVIEW

File: table/table_test.go [hunk 1]
```diff
@@ -725,7 +725,7 @@ func TestTableWidthSmartCrop(t *testing.T) {
 
 	expected := strings.TrimSpace(`
 ┌──────┬─────┬──────────┐
-│ Name │ Age │ Location │
```
Comment (maaslalani) [inline table/table_test.go:728]: I believe this is incorrect. It should be `Age` not `Ag…`

File: table/table_test.go [hunk 2]
```diff
@@ -757,15 +757,16 @@ func TestTableWidthSmartCropExtensive(t *testing.T) {
...
 ┣━━━━╋━━━━━╋━━━━━┫
-┃ Ch ┃ 您  ┃ 你  ┃
-┃ Ja ┃ こ  ┃ や  ┃
-┃ Ar ┃ أهل ┃ أهل ┃
-┃ Ru ┃ Здр ┃ При ┃
-┃ Sp ┃ Hol ┃ ¿Qu ┃
-┃ En ┃ You ┃ How ┃
+┃ C… ┃ 您… ┃ 你… ┃
+┃ J… ┃ こ… ┃ や… ┃
+┃ A… ┃ أه… ┃ أه… ┃
```
Comment (maaslalani) [inline table/table_test.go:764]: I believe this line is incorrect.

**Comment by maaslalani** [comment]
I might be incorrect but I think I decision to not prioritize padding over text content. This is important when resizing, so that we remove padding before truncating content.

**Comment by caarlos0** [comment]
yeah this was actually another issue

**Pull request closed without merging**